storage-node/src/services/helpers/acceptObject.ts

```diff
@@ -0,0 +1,28 @@
+import fs from 'fs'
+import logger from '../logger'
+import { getStorageProviderConnection, isStorageProviderConnectionEnabled } from '../../commands/server'
+import { addDataObjectIdToCache } from '../caching/localDataObjects'
+import { moveFile } from './moveFile'
+const fsPromises = fs.promises
+
+export async function acceptObject(
+  filename: string,
+  src: fs.PathLike,
+  dest: fs.PathLike | undefined = undefined
+): Promise<void> {
+  const toBeAcceptedOnLocalVolume = !isStorageProviderConnectionEnabled()
+  if (toBeAcceptedOnLocalVolume) {
+    if (dest === undefined) {
+      throw new Error('Destination path is undefined')
+    }
+    await moveFile(src, dest)
+    logger.info(`File ${filename} accepted on local volume`)
+  } else {
+    const connection = getStorageProviderConnection()!
+    await connection.uploadFileToRemoteBucket(filename, src.toString())
+    await fsPromises.unlink(src)
+    logger.info(`File ${filename} accepted to remote storage`)
+  }
+  addDataObjectIdToCache(filename)
```

Comment:
Missing the second argument to indicate where object is stored.
```suggestion
  addDataObjectIdToCache(filename, toBeAcceptedOnLocalVolume)
```

Reply:
For both moveFile and uploadToRemoteBucket should we explicitly handle case if destination object already exists?